crates/conformance-test-utils/src/madara.rs

```diff
@@ -0,0 +1,163 @@
+    const COUNTER_DEPLOYED_BYTECODE_LEN: u64 = (COUNTER_DEPLOYED_BYTECODE.len() / 2 - 1) as u64; // remove 0x and divide by 2 for byte length
+
+    use super::*;
+
+    #[test]
+    fn test_genesis_load_bytecode() {
+        // Given
+        let bytecode = Bytes::from_str(TEST_BYTECODE).unwrap();
+        let address = *ACCOUNT_ADDRESS;
+
+        // When
+        let storage = genesis_load_bytecode(bytecode, address);
+
+        // Then
+        let expected_storage: Vec<((Felt, Felt), Felt)> = vec![
+            (
+                (address.into(), get_storage_var_address("bytecode_", &[FieldElement::from(0u8)]).unwrap().into()),
+                FieldElement::from_hex_be(BIG_ENDIAN_BYTECODE).unwrap().into(),
+            ),
+            (
+                (address.into(), get_storage_var_address("bytecode_", &[FieldElement::from(1u8)]).unwrap().into()),
+                FieldElement::from_hex_be(BIG_ENDIAN_BYTECODE).unwrap().into(),
+            ),
+        ];
+        assert_eq!(expected_storage, storage);
+    }
+
+    #[tokio::test]
+    async fn test_counter_bytecode() {
+        // Construct a Starknet test sequencer
+        let starknet_test_sequencer = construct_kakarot_test_sequencer().await;
+
+        // Define the expected funded amount for the Kakarot system
+        let expected_funded_amount = FieldElement::from_dec_str("1000000000000000000").unwrap();
+
+        // Deploy the Kakarot system
+        let deployed_kakarot =
+            deploy_kakarot_system(&starknet_test_sequencer, EOA_WALLET.clone(), expected_funded_amount).await;
+
+        // Deploy a counter contract
+        let (_, deployed_addresses) = deployed_kakarot
+            .deploy_evm_contract(
+                starknet_test_sequencer.url(),
+                "Counter",
+                // no constructor is conveyed as a tuple
+                (),
+            )
+            .await
+            .unwrap();
+        let expected_counter_address = deployed_addresses[1]; // index 0 is evm address, index 1 is starknet address
+
+        // Deploy a safe contract for which we will set the counter bytecode
```

Comment:
I don't understand why we need to deploy a `Safe` contract. Like, we deploy a true contract, but hard replace it's bytecode in the starknet object and check that eventually it has become a counter, isn't it weird?

Can't we just use a random target address, put the storage accordingly, and use directly

```
        let actual_counter = ContractAccount::new(&starknet_client, actual_counter_address);
```

Reply:
I need to check if this is possible, the reason I did this is that I believe the test sequencer needs to know have in it's internal mapping that some contract exists at the address. I guess I could play with the state and just set the contract address and the class hash manually.